plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,336 @@
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{}
+	})
+}
+
+func parseMetric(defaultHostname, defaultServiceState, resourceTag string, metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := defaultHostname
+
+	if value, present := metric.GetTag(resourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := defaultServiceState
+	if value, present := metric.GetTag("status"); present {
+		if validStatus(value) {
+			status = value
+		}
+	}
+
+	message := ""
+	if value, present := metric.GetTag("message"); present {
+		message = value
+	}
+
+	unitType := string(transit.UnitCounter)
+	if value, present := metric.GetTag("unitType"); present {
+		unitType = value
+	}
+
+	critical := -1.0
+	if value, present := metric.GetTag("critical"); present {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			critical = s
+		}
+	}
+
+	warning := -1.0
+	if value, present := metric.GetTag("warning"); present {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			warning = s
+		}
+	}
+
+	serviceObject := transit.DynamicMonitoredService{
+		BaseTransitData: transit.BaseTransitData{
+			Name:  service,
+			Type:  transit.Service,
+			Owner: resource,
+		},
+		Status:           transit.MonitorStatus(status),
+		LastCheckTime:    milliseconds.MillisecondTimestamp{Time: time.Now()},
+		NextCheckTime:    milliseconds.MillisecondTimestamp{},
+		LastPlugInOutput: message,
+		Metrics:          nil,
+	}
+
+	for _, value := range metric.FieldList() {
+		var thresholds []transit.ThresholdValue
+		thresholds = append(thresholds, transit.ThresholdValue{
+			SampleType: transit.Warning,
+			Label:      value.Key + "_wn",
+			Value: &transit.TypedValue{
+				ValueType:   transit.DoubleType,
+				DoubleValue: warning,
+			},
+		})
+		thresholds = append(thresholds, transit.ThresholdValue{
+			SampleType: transit.Critical,
+			Label:      value.Key + "_cr",
+			Value: &transit.TypedValue{
+				ValueType:   transit.DoubleType,
+				DoubleValue: critical,
+			},
+		})
+
+		valueType := transit.DoubleType
+		var floatVal float64
+		var stringVal string
+
+		switch value.Value.(type) {
+		case string:
+			valueType = transit.StringType
+			tmpStr := value.Value.(string)
+			stringVal = tmpStr
+		default:
+			floatVal, _ = internal.ToFloat64(value.Value)
+		}
+		serviceObject.Metrics = append(serviceObject.Metrics, transit.TimeSeries{
+			MetricName: value.Key,
+			SampleType: transit.Value,
+			Interval: &transit.TimeInterval{
+				EndTime: milliseconds.MillisecondTimestamp{Time: metric.Time()},
+			},
+			Value: &transit.TypedValue{
+				ValueType:   valueType,
+				DoubleValue: floatVal,
+				StringValue: stringVal,
+			},
+			Unit:       transit.UnitType(unitType),
+			Thresholds: &thresholds,
+		})
+	}
+
+	serviceObject.Status, _ = transit.CalculateServiceStatus(&serviceObject.Metrics)
```

Comment:
Could this be done only if `status` tag is not present?

Reply:
Done